### PR TITLE
tg-can-1183

### PR DESCRIPTION
**Pull request by nalwa**
This PR has fixes for can-1183, can-1184, can-1188, can-1189

@akuks  please review and merge PR

### CODE REVIEW

File: app/Helpers/Util.php [hunk 1]
```diff
@@ -392,17 +392,36 @@ public function checkParentCampChanged($all, $in_review_status, $liveCamp)
...
         if($all['parent_camp_num'] != $all['old_parent_camp_num']) {
                 $allParentCamps = Camp::getAllParent($liveCamp);
-                $allChildSupporters = Support::where('topic_num', $all['topic_num'])
+                $supporterNicknames = Support::where('topic_num', $all['topic_num'])
                     ->where('end', 0)
                     ->whereIn('camp_num', $allParentCamps)
                     ->pluck('nick_name_id');
-                if (sizeof($allChildSupporters) > 0) {
+
+                    if (sizeof($supporterNicknames) > 0)
```
Comment (akuks) [inline app/Helpers/Util.php]: `{` should be inline with others if and foreach statement

**Review by akuks — CHANGES_REQUESTED** [review]

File: app/Helpers/Util.php (hunk 1 above)
Comment (nalwa) [reply app/Helpers/Util.php]: done

File: app/Helpers/Util.php [hunk 2]
```diff
@@ -392,17 +392,36 @@ public function checkParentCampChanged($all, $in_review_status, $liveCamp)
...
-                $allChildSupporters = Support::where('topic_num', $all['topic_num'])
+                $supporterNicknames = Support::where('topic_num', $all['topic_num'])
                     ->where('end', 0)
                     ->whereIn('camp_num', $allParentCamps)
                     ->pluck('nick_name_id');
-                if (sizeof($allChildSupporters) > 0) {
-                    foreach ($allParentCamps as $parentCamp) {
-                        Support::removeSupport($all['topic_num'], $parentCamp, $allChildSupporters, $all['camp_num']);
+
+                if (sizeof($supporterNicknames) > 0)
```
Comment (akuks) [inline app/Helpers/Util.php]: Please align `if(){` as in the line 396. Similar foreach in the below statements

File: app/Helpers/Util.php [hunk 3]
```diff
@@ -392,17 +392,36 @@ public function checkParentCampChanged($all, $in_review_status, $liveCamp)
...
+                    {
+                        foreach($supporterNicknames as $nickId)
+                        {
+                            Support::removeSupportWithDelegates($all['topic_num'], $parentCamp, $nickId); 
+                            Support::reOrderSupport($all['topic_num'], [$nickId]);
+                        }
                     }
                 }
+
+               /* if (sizeof($supporterNicknames) > 0) {
```
Comment (akuks) [inline app/Helpers/Util.php:412]: Remove this if not required

**Review by akuks — CHANGES_REQUESTED** [review]
@nalwa Please do the required changes

**Review by akuks — APPROVED** [review]

**Pull request merged**